### ISSUE TITLE
Increased Keyboard Backlight Levels/Steps using Windows Dynamic Lighting (API)?

### ISSUE THREAD

**Issue by nathancdy** [seerge/g-helper#2380]
### Rules

- [X] I made myself familiar with the <a href='https://github.com/seerge/g-helper?tab=readme-ov-file'>Readme</a>, <a href='https://github.com/seerge/g-helper/wiki/FAQ'>FAQ</a> and <a href='https://github.com/seerge/g-helper/wiki/Troubleshooting'>Troubleshooting</a>.
- [X] I understand that, if insufficient information will be provided, my issue will be closed without an answer.

### Is your feature request related to a problem? Please describe

ASUS has very few steps/levels for their keyboard backlight, some people may want to a more precise brightness level for the keyboard backlight.

### Describe the solution you'd like

(I haven't tested myself, but I've heard other accounts of people using the Windows Dynamic Lighting to have more precise levels of keyboard backlighting)

If there is a way to implement the Windows Dynamic Lighting to increase the number of levels or steps for the keyboard backlighting, similar to how MacBooks implement their keyboard backlight levels.

### Device and Model

ASUS Zephyrus G14 GA403UV

### Additional information.

I'm sure this has been requested in the past though I haven't found the request myself. Feel free to close this request if it has already been addressed. Thanks!

**Comment by seerge** [comment]
@nathancdy 

Ok. We need to figure out what exactly Armoury sends to keyboard.

Can you please do following

Install https://www.hhdsoftware.com/device-monitoring-studio (will require reboot)

1. Launch Monitoring Studio, pick Asus device from the list and double click on it
![Screenshot 2024-01-20 163125](https://github.com/seerge/g-helper/assets/5920850/e9a28802-08cb-423c-812b-7b526b051a5f)
2. Add ``Packet View`` to Procesing and click ``Start``
![Screenshot 2024-01-20 163136](https://github.com/seerge/g-helper/assets/5920850/354340f0-2fcb-473b-a49f-2e01b25d00a9)
3. Change backlight brightness using Windows Dynamic Lightning
4. Check if Monitoring Studio catching a lot of data (Armoury spams it like crazy). If NOT - you probably have selected wrong Asus device on step 2 (so need to re-do it)
5. In Monitoring Studio select ``Tools`` -> ``Save to Log``, and save data
![Screenshot 2024-01-20 163247](https://github.com/seerge/g-helper/assets/5920850/ad480dcf-05bc-43a7-9a80-2984fb68e8a8)

Upload this data here.

**Comment by nathancdy** [comment]
@seerge will try to do some testing once i am home!

Does the communication between AC and the keyboard differ across year models? I assume you may have already done some testing with your G14s but need someone with the 2024 model

**Comment by seerge** [comment]
@nathancdy no, it's practically same across all year. Except G15 2022 is broken and has some own commands. 

But it could be that "new" approach of windows dynamic lightning is different tho

**Comment by nathancdy** [comment]
@seerge Not home at the moment, but I noticed in my Human Interface Devices section in Device Manager, I don't have an ASUS-labeled item on the list. The only ASUS-labeled item is the Precision Touchpad and the rest are HID _____. 

![image](https://github.com/seerge/g-helper/assets/103626631/0e276a97-43c9-4c01-8199-e71f53162617)

**Comment by seerge** [comment]
@nathancdy any updates on the matter ?

**Comment by nathancdy** [comment]
@seerge sorry, was busy with school initially during the time I sent my last reply and just seen your reply just now. Should I be testing each device in Human Interface Devices? I'm unsure why I don't have an ASUS-labeled device under that section.
